### DISCUSSION TITLE
Seeking Guidance on Writing Unit Test Cases for JustPy Application

### DISCUSSION THREAD

**Discussion by sanket-parte** [elimintz/justpy#671] [General]
Hi, I would like to write unit test cases for my JustPy application. Can anyone suggest how I can accomplish this?

**Comment by WolfgangFahl** [comment]
See the unit tests of justpy itself. Please note that async needs extra care and for end-to end tests you might need selenium. Also stackoverflow is your friend - the community there is larger than here.
